src/gfnff/gfnff_eg.f90

```diff
@@ -0,0 +1,3218 @@
+         term2(1:3)=ea*damp2jk*dampij*vcb(1:3)
+         g(1:3,1) = -dedb(1:3)*damp-term1(1:3)-term2(1:3)
+         g(1:3,2) =  deda(1:3)*damp+term1(1:3)
+         g(1:3,3) =  dedc(1:3)*damp+term2(1:3)
+
+      end
+
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+      subroutine egbend_nci_mul(j,i,k,c0,fc,n,at,xyz,e,g)
+      use gff_param
+      implicit none
+      !Dummy
+      integer n,at(n)
+      integer i,j,k
+      real*8  c0,fc
+      real*8  xyz(3,n),g(3,3),e
+      !Stack
+      real*8  kijk,va(3),vb(3),vc(3),cosa
+      real*8  dt,ea,dedb(3),dedc(3),rmul2,rmul1,deddt
+      real*8  term1(3),term2(3),rab2,vab(3),vcb(3),rp
+      real*8  rcb2,damp,dampij,damp2ij,dampjk,damp2jk
+      real*8  theta,pi,deda(3),vlen,vp(3),et,dij,c1
+      real*8  term3(3),x1sin,x1cos,e1,dphi1,vdc(3)
+      real*8  ddd(3),ddc(3),ddb(3),dda(3),rjl,phi
+      real*8  omega,rij,rijk,phi0,rkl,rjk,dampkl,damp2kl
+      real*8  dampjl,damp2jl,valijklff,rn
+      parameter (pi = 3.1415926535897932384626433832795029d0)
+
+
+         kijk=fc/(cos(0.0d0)-cos(c0))**2
+         va(1:3) = xyz(1:3,i)
+         vb(1:3) = xyz(1:3,j)
+         vc(1:3) = xyz(1:3,k)
+         call vsub(va,vb,vab,3)
+         call vsub(vc,vb,vcb,3)
+         rab2 = vab(1)*vab(1) + vab(2)*vab(2) + vab(3)*vab(3)
+         rcb2 = vcb(1)*vcb(1) + vcb(2)*vcb(2) + vcb(3)*vcb(3)
+         call crprod(vcb,vab,vp)
+         rp = vlen(vp)+1.d-14
+         call impsc(vab,vcb,cosa)
+         cosa = dble(min(1.0d0,max(-1.0d0,cosa)))
+         theta= dacos(cosa)
+
+         if(pi-c0.lt.1.d-6)then     ! linear
+         dt  = theta - c0
+         ea  = kijk * dt**2
+         deddt = 2.d0 * kijk * dt
+         else
+         ea=kijk*(cosa-cos(c0))**2  ! not linear
+         deddt=2.0d0*kijk*sin(theta)*(cos(c0)-cosa)
+         endif
+
+         e = (1.0d0-ea)
+         call crprod(vab,vp,deda)
+         rmul1 = -deddt / (rab2*rp)
+         deda = deda*rmul1
+         call crprod(vcb,vp,dedc)
+         rmul2 =  deddt / (rcb2*rp)
+         dedc = dedc*rmul2
+         dedb = deda+dedc
+         g(1:3,1) =  dedb(1:3)
+         g(1:3,2) = -deda(1:3)
+         g(1:3,3) = -dedc(1:3)
+
+      end
+
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+      subroutine egbend_nci(j,i,k,c0,kijk,n,at,xyz,e,g)
+      use gff_param
+      implicit none
+      !Dummy
+      integer n,at(n)
+      integer i,j,k
+      real*8 c0,kijk
+      real*8 xyz(3,n),g(3,3),e
+      !Stack
+      real*8  va(3),vb(3),vc(3),cosa
+      real*8  dt,ea,dedb(3),dedc(3),rmul2,rmul1,deddt
+      real*8  term1(3),term2(3),rab2,vab(3),vcb(3),rp
+      real*8  rcb2,damp,dampij,damp2ij,dampjk,damp2jk
+      real*8  theta,pi,deda(3),vlen,vp(3),et,dij,c1
+      real*8  term3(3),x1sin,x1cos,e1,dphi1,vdc(3)
+      real*8  ddd(3),ddc(3),ddb(3),dda(3),rjl,phi
+      real*8  omega,rij,rijk,phi0,rkl,rjk,dampkl,damp2kl
+      real*8  dampjl,damp2jl,valijklff,rn
+      parameter (pi = 3.1415926535897932384626433832795029d0)
```

Comment:
We have this one in `xtb_mctc_constants`

Reply:
now using `xtb_mctc_constants`